File: packages/core/tests/fixtures/unhealthy/crypto-misuse.py
# Unhealthy fixture: cryptographic misuse in Python
# Sprint 58 — detectSecuritySinks fixture
import hashlib
import random
import secrets  # safe alternative — for reference

# CryptographicMisuseRisk: hashlib.md5 used for password hashing
def hash_password(password: str) -> str:
    return hashlib.md5(password.encode()).hexdigest()

# CryptographicMisuseRisk: random.random() used as cryptographic nonce
def generate_nonce() -> float:
    nonce = random.random()
    return nonce

# CryptographicMisuseRisk: hashlib.sha1 used for token generation
def compute_token(data: str) -> str:
    return hashlib.sha1(data.encode()).hexdigest()

# Safe usage — should NOT trigger
def secure_token() -> str:
    return secrets.token_hex(32)

File: packages/core/tests/fixtures/unhealthy/crypto-misuse.ts
// Unhealthy fixture: cryptographic misuse patterns
// Sprint 58 — detectSecuritySinks fixture
import { createHash, randomBytes } from 'crypto';

// CryptographicMisuseRisk: MD5 used for security-critical hashing
function hashPassword(password: string): string {
  return createHash('md5').update(password).digest('hex');
}

// CryptographicMisuseRisk: Math.random() used for security token generation
function generateToken(): string {
  const token = Math.random().toString(36).slice(2);
  return token;
}

// CryptographicMisuseRisk: SHA-1 for signing
function signData(data: string): string {
  return createHash('sha1').update(data).digest('hex');
}

// Safe usage — should NOT trigger
function secureToken(): string {
  return randomBytes(32).toString('hex');
}
